clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -952,6 +954,32 @@ public String toString() {
         }
     }
 
+    public static class LambdaValidator implements Validator {
+        BiConsumer<String, Object> ensureValid;
+        Supplier<String> toStringFunction;
+
+        private LambdaValidator(BiConsumer<String, Object> ensureValid,
+                                Supplier<String> toStringFunction) {
+            this.ensureValid = ensureValid;
+            this.toStringFunction = toStringFunction;
+        }
+
+        public static LambdaValidator with(BiConsumer<String, Object> ensureValid,
+                                           Supplier<String> toStringFunction) {
+            return new LambdaValidator(ensureValid, toStringFunction);
+        }
+
+        @Override
+        public void ensureValid(String name, Object value) {
+            ensureValid.accept(name, value);
```

Comment:
It's strange to rely on RuntimeExceptions for validation like this. Maybe narrow to ConfigExceptions at least.

Reply:
Again, this is added for convenience. It's not implementing the validation as other validators do in this class. What developers chooses to throw here, is their own responsibility, as when they implement the `Validator` interface.